htdocs/core/db/DoliDB.class.php

```diff
@@ -411,4 +436,27 @@ public function getRows($sql)
 
 		return false;
 	}
+
+	/**
+	 * Return a column from query as an array.
+	 * Only to be used with SELECT queries.
+	 *
+	 * @param 	string 		$sql 		The sql query string
+	 * @return 	bool|array				Result
+	 */
+	public function getCol($sql)
+	{
+		$res = $this->query($sql);
+		if ($res) {
+			$results = array();
+			if ($this->num_rows($res) > 0) {
+				while ($arr = $this->fetch_row($res)) {
```

Comment:
Using of fetch_row make unnamed access to data.
Column access must use a named access to avoid bug or bad data use when field orders differs from one database to another one.

Reply:
The idea here is to use with queries like:

```
SELECT nom FROM d_projet;
```

to get a list of names as an array, where only one column is being selected, so that field order is not applicable.  The same applies to both `getVar()` and `getCol()`.